src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -158,7 +161,7 @@ def attach_dataloaders(
         self.trainer.train_dataloader = None
         self.trainer.val_dataloaders = None
         self.trainer.test_dataloaders = None
-        self.trainer.predict_dataloaders = None
+        self.trainer.predict_dataloaders = None  # type: ignore[assignment]
```

Comment:
I think adding an entry here:
https://github.com/Lightning-AI/lightning/blob/ee4233b36cb70b19a13013371ada50875b429ef3/src/pytorch_lightning/trainer/trainer.py#L634
```py
self.predict_dataloaders = None
```

will help avoid `ignore[assignment]` here.

Reply:
doing this causes new errors 

```sh
src/pytorch_lightning/loops/dataloader/prediction_loop.py:63: error: Incompatible return value type (got "None", expected "Sequence[DataLoader[Any]]")  [return-value]
```

above error can be resolved with

```python
def dataloaders(self) -> Sequence[DataLoader]:
    """Returns all prediction dataloaders."""
    if self.trainer.predict_dataloaders:
        return self.trainer.predict_dataloaders
```

is it appropriate to annotate in Trainer, and update logic in a third file in order to remove an ignore?

Reply:
added a commit. removed 4 `ignores`